src/Hid.Shared/Hid.cs

```diff
@@ -0,0 +1,174 @@
+        [DllImport(nameof(Hid), SetLastError = true)]
+        public static extern bool HidD_GetAttributes(
+            SafeFileHandle hidDeviceObject,
+            ref HiddAttributes attributes);
+
+        /// <summary>
+        /// Returns a top-level collection's embedded string that identifies the manufacturer.
+        /// </summary>
+        /// <param name="hidDeviceObject">Specifies an open handle to a top-level collection.</param>
+        /// <param name="buffer">
+        /// Pointer to a caller-allocated buffer that the routine uses to return the collection's manufacturer
+        /// string.
+        /// </param>
+        /// <param name="bufferLength">
+        /// Specifies the length, in bytes, of a caller-allocated buffer provided at Buffer. If the
+        /// buffer is not large enough to return the entire NULL-terminated embedded string, the routine returns nothing in the
+        /// buffer.
+        /// </param>
+        /// <returns>TRUE if it returns the entire NULL-terminated embedded string. Otherwise, the routine returns FALSE.</returns>
+        /// <remarks>
+        /// The maximum possible number of characters in an embedded string is device specific. For USB devices, the
+        /// maximum string length is 126 wide characters (not including the terminating NULL character).
+        /// </remarks>
+        [DllImport(nameof(Hid), CharSet = CharSet.Auto, SetLastError = true)]
+        public static extern bool HidD_GetManufacturerString(
+            SafeFileHandle hidDeviceObject,
+            StringBuilder buffer,
+            int bufferLength);
+
+        /// <summary>
+        /// Returns the embedded string of a top-level collection that identifies the manufacturer's product.
+        /// </summary>
+        /// <param name="hidDeviceObject">Specifies an open handle to a top-level collection.</param>
+        /// <param name="buffer">Pointer to a caller-allocated buffer that the routine uses to return the requested product string.</param>
+        /// <param name="bufferLength">
+        /// Specifies the length, in bytes, of a caller-allocated buffer provided at Buffer. If the
+        /// buffer is not large enough to return the entire NULL-terminated embedded string, the routine returns nothing in the
+        /// buffer.
+        /// </param>
+        /// <returns>
+        /// TRUE if it successfully returns the entire NULL-terminated embedded string. Otherwise, the routine returns
+        /// FALSE.
+        /// </returns>
+        /// ///
+        /// <remarks>
+        /// The maximum possible number of characters in an embedded string is device specific. For USB devices, the
+        /// maximum string length is 126 wide characters (not including the terminating NULL character).
+        /// </remarks>
+        [DllImport(nameof(Hid), CharSet = CharSet.Unicode, SetLastError = true)]
+        public static extern bool HidD_GetProductString(
+            SafeFileHandle hidDeviceObject,
```

Comment:
When this type is no longer available (for portable), use the `SafeObjectHandle` type defined in Kernel32. You'll have to add a project reference to that from your desktop and portable projects, and add a project reference from you nuproj to the Kernel32 nuproj